lib/pavilion/scriptcomposer.py

```diff
@@ -0,0 +1,697 @@
+        self.__init__()
+
+
+class scriptCommands( object ):
+    """Class to contain the script commands."""
+
+    @property
+    def commands( self ):
+        return self._commands
+
+    @commands.setter
+    def commands( self, value ):
+        """Function to set the internal commands variable."""
+        if value != None and not isinstance( value, list ):
+            error = "Commands must be of type 'list' and not " +\
+                    "{}.".format( type( value ) )
+            raise TypeError( error )
+        self._commands = value
+
+    def __init__( self, commands=None ):
+        """Function to specify the commands for the script.
+        :param list commands: List of strings specifying the script commands
+                              in order.  default = None
+        """
+        self.commands = commands
+
+    def reset( self ):
+        self.__init__()
+
+
+class scriptPost( object ):
+    """Class to contain the post-script commands."""
+
+    @property
+    def commands( self ):
+        return self._commands
+
+    @commands.setter
+    def commands( self, value ):
+        """Function to set the internal commands variable."""
+        if value != None and not isinstance( value, list ):
+            error = "Commands must be of type 'list' and not " +\
+                    "{}.".format( type( value ) )
+            raise TypeError( error )
+        self._commands = value
+
+    def __init__( self, commands=None ):
+        """Function to specify the commands to run at the end of the script
+        for generic tasks.
+        :param list commands: List of strings specifying the postscript
+                              commands in order.  default = None
+        """
+        self.commands = commands
+
+    def reset( self ):
+        self.__init__()
+
+
+class scriptDetails( object ):
+    """Class to contain the final details of the script."""
+
+    @property
+    def name( self ):
+        return self._name
+
+    @name.setter
+    def name( self, value ):
+        if not isinstance( value, unicode ):
+            error = "Name must be of type 'unicode' and not " +\
+                    "{}.".format( type( value ) )
+            raise TypeError( error )
+        self._name = value
+
+    @property
+    def script_type( self ):
+        return self._script_type
+
+    @script_type.setter
+    def script_type( self, value ):
+        if not isinstance( value, unicode ):
+            error = "Script type must be of type 'unicode' and not " +\
+                    "{}.".format( type( value ) )
+            raise TypeError( error )
+        self._script_type = value
+
+    @property
+    def user( self ):
+        return self._user
+
+    @user.setter
+    def user( self, value ):
+        if not isinstance( value, unicode ):
+            error = "User must be of type 'unicode' and not " +\
+                    "{}.".format( type( value ) )
+            raise TypeError( error )
+        self._user = value
+
```

Comment:
All this group and permission's stuff is really convoluted.  Why not just have it be an init arg?

Reply:
I don't understand what you mean.  It is an init arg with what I think are sensible defaults.  I'm just trying to balance between being flexible enough to not be LANL-specific and yet provide reasonable checks.  It can also make the set of classes more generally applicable outside of pavilion.